jps-plugin/src/org/jetbrains/jps/incremental/scala/data/CompilationData.scala

```diff
@@ -81,7 +81,10 @@ object CompilationData {
       val scalaVersion = CompilerData.compilerVersion(module)
       val hydraOptions =
         if (hydraSettings.isHydraEnabled && scalaVersion.nonEmpty && hydraGlobalSettings.containsArtifactsFor(scalaVersion.get, hydraSettings.getHydraVersion))
-          Seq("-sourcepath", outputGroups.map(_._1).mkString(File.pathSeparator), "-cpus", "2")
+          Seq("-sourcepath", outputGroups.map(_._1).mkString(File.pathSeparator), "-cpus", hydraSettings.getNumberOfCores,
+            "-YsourcePartitioner:" + hydraSettings.getSourcePartitioner, "-YhydraStore", hydraSettings.getHydraStorePath,
+            "-YpartitionFile", hydraSettings.getHydraStorePath + module.getName, "-YrootDirectory", hydraSettings.getProjectRoot,
+            "-YtimingsFile", hydraSettings.getHydraStorePath + "timings.csv", "-YhydraTag", module.getName)
```

Comment:
I guess `getHydraStorePath` is guaranteed to end with a proper file separator? "/" or "\"?